front/components/assistant_builder/ActionsScreen.tsx

```diff
@@ -608,3 +623,57 @@ function ActionEditor({
     </div>
   );
 }
+
+function AdvancedSettings({
+  maxToolsUsePerRun,
+  setMaxToolsUsePerRun,
+}: {
+  maxToolsUsePerRun: number | null;
+  setMaxToolsUsePerRun: (maxToolsUsePerRun: number | null) => void;
+}) {
+  return (
+    <DropdownMenu>
+      <DropdownMenu.Button>
+        <Button
+          label="Advanced settings"
+          variant="tertiary"
+          size="sm"
+          type="menu"
+        />
+      </DropdownMenu.Button>
+      <DropdownMenu.Items width={240} overflow="visible">
+        <div className="flex flex-col gap-4">
+          <div className="flex flex-col gap-2">
+            <div className="flex flex-col items-start justify-start">
+              <div className="w-full grow text-sm font-bold text-element-800">
+                Max actions per run
+              </div>
+              <div className="w-full grow text-sm text-element-600">
+                up to {MAX_TOOLS_USE_PER_RUN_LIMIT}
+              </div>
+            </div>
+            <Input
+              value={maxToolsUsePerRun?.toString() ?? ""}
```

Comment:
We don't have type on Input?

Reply:
Would be nice to add. A text input for a number always looks broken-ish

Reply:
Def agreed 👍